desc/plotting.py

```diff
@@ -492,21 +531,24 @@ def plot_2d(eq, name, grid=None, log=False, norm_F=False, ax=None, **kwargs):
         data = np.abs(data)  # ensure data is positive for log plot
         contourf_kwargs["norm"] = matplotlib.colors.LogNorm()
         if norm_F:
-            contourf_kwargs["levels"] = kwargs.get("levels", np.logspace(-6, 0, 7))
+            contourf_kwargs["levels"] = kwargs.pop("levels", np.logspace(-6, 0, 7))
         else:
             logmin = max(np.floor(np.nanmin(np.log10(data))).astype(int), -16)
             logmax = np.ceil(np.nanmax(np.log10(data))).astype(int)
-            contourf_kwargs["levels"] = kwargs.get(
+            contourf_kwargs["levels"] = kwargs.pop(
                 "levels", np.logspace(logmin, logmax, logmax - logmin + 1)
             )
     else:
         contourf_kwargs["norm"] = matplotlib.colors.Normalize()
-        contourf_kwargs["levels"] = kwargs.get(
+        contourf_kwargs["levels"] = kwargs.pop(
             "levels", np.linspace(np.nanmin(data), np.nanmax(data), 100)
         )
-    contourf_kwargs["cmap"] = kwargs.get("cmap", "jet")
+    contourf_kwargs["cmap"] = kwargs.pop("cmap", "jet")
     contourf_kwargs["extend"] = "both"
 
+    if len(kwargs):
+        raise ValueError(f"plot_2d got unexpected keyword argument: {kwargs.keys()}")
```

Comment:
a better way to handle this might be ``assert len(kwargs)==0, msg``, that way it will always be checked by the testing